### DISCUSSION TITLE
Should we fix all issues on Dart analyze before create PR?

### DISCUSSION THREAD

**Discussion by aldychris** [serverpod/serverpod#303] [Q&A] [answered]
When I try to run dart analyser from root folder it telling me 

`error: Undefined name 'PRODUCTION_MODE'. (undefined_identifier at [%unnamed%] bin/generated/version.dart:5)`

This one is generated file so how to regenerate this const. 

Other thing is when I try run the analyser its give me 18 warnings and 206 hints, when we creating a PR should we resolve all issue or just need to fix the error one?

Is there any script in util folder for me to check all process before I create PR?

Thanks

**Comment by vlidholt** [comment, marked as answer]
I don't think the dart analyzer will be happy from the root folder. However, there are scripts to run them on every individual package, `util/run_tests_analyze`. All packages are analyzed by the CI on the Serverpod repo.
